src/AppInstallerCLIE2ETests/InstallCommand.cs

```diff
@@ -155,6 +163,65 @@ public void InstallExeWithAlternateSourceFailure()
             }
         }
 
+        [Test]
+        public void InstallPortableExe()
+        {
+            var installDir = TestCommon.GetRandomTestDir();
+            string packageId, commandAlias, fileName, productCode;
+            packageId = commandAlias = productCode = "AppInstallerTest.TestPortableExe";
+            fileName = "AppInstallerTestExeInstaller.exe";
+
+            var result = TestCommon.RunAICLICommand("install", $"{packageId} -l {installDir}");
+            Assert.AreEqual(Constants.ErrorCode.S_OK, result.ExitCode);
+            Assert.True(result.StdOut.Contains("Successfully installed"));
+            Assert.True(VerifyTestPortableInstalledAndCleanup(installDir, packageId, commandAlias, fileName, productCode));
+        }
+
+        [Test]
+        public void InstallPortableExeWithCommand()
+        {
+            var installDir = TestCommon.GetRandomTestDir();
+            string packageId, commandAlias, fileName, productCode;
+            packageId = productCode = "AppInstallerTest.TestPortableExeWithCommand";
+            fileName = "AppInstallerTestExeInstaller.exe";
+            commandAlias = "testCommand";
+
+            var result = TestCommon.RunAICLICommand("install", $"{packageId} -l {installDir}");
+            Assert.AreEqual(Constants.ErrorCode.S_OK, result.ExitCode);
+            Assert.True(result.StdOut.Contains("Successfully installed"));
+            Assert.True(VerifyTestPortableInstalledAndCleanup(installDir, packageId, commandAlias, fileName, productCode));
+        }
+
+        [Test]
+        public void InstallPortableExeWithAppsAndFeatures()
+        {
+            var installDir = TestCommon.GetRandomTestDir();
+            string packageId, commandAlias, fileName, productCode;
+            packageId = "AppInstallerTest.TestPortableExeWithAppsAndFeatures";
+            productCode = "testProductCode";
+            commandAlias = fileName = "AppInstallerTestExeInstaller.exe";
+
+            var result = TestCommon.RunAICLICommand("install", $"{packageId} -l {installDir}");
+            Assert.AreEqual(Constants.ErrorCode.S_OK, result.ExitCode);
+            Assert.True(result.StdOut.Contains("Successfully installed"));
+            Assert.True(VerifyTestPortableInstalledAndCleanup(installDir, packageId, commandAlias, fileName, productCode));
+        }
+
+        [Test]
+        public void InstallPortableExeWithRename()
+        {
+            var installDir = TestCommon.GetRandomTestDir();
+            string packageId, productCode, renameArgValue;
+            packageId = "AppInstallerTest.TestPortableExeWithRename";
+            productCode = "testProductCode";
+            renameArgValue = "testRename";

```

Comment:
Can you add a test for an invalid filename like `testDisallowedChars?` and `CON` ?

Reply:
Added two additional E2E tests for disallowedChars and reservednames and explicitly handle those cases in the implementation